### ISSUE TITLE
Firefox status bar hiding input line

### ISSUE THREAD

**Issue by Massimo-B** [tridactyl/tridactyl#4223]
-   Brief description of the problem:
Sometimes the Firefox statusbar is hiding the VIM input line of Tridactyl.

-   Steps to reproduce:
Usually I press t to open a new tab, then start typing part of the name. Sometimes the Firefox statusbar appears above the : input line and hides the typing. I can't really reproduce when and how the statusbar appears. It has something to do with the pointer hovering at a hyperlink. If the hyperlink is already hovered at before starting to type a : command, the statusbar disappears as expected. Maybe the pointer is moved by mistake while typing. Can you somehow prevent the statusbar appearing while the input line is active?

-   Tridactyl version (`:version`):
1.22.1pre6200-21b7546b

-   Firefox version (Top right menu > Help > About Firefox):
91.8.0

**Comment by bovine3dom** [comment]
there's `:guiset hoverlink none` if you have the native messenger installed, which should get rid of the little pop-ups permanently.

Otherwise you could just use a theme with the commandline on top like `:colours quakelight`, which is what I personally use :)

**Comment by Massimo-B** [comment]
`:colours quakelight` is a workaround, though I need to get used to it after half a life using vim.
Why is that theme showing the ] character when starting with : key? Just a theme decision?

**Comment by bovine3dom** [comment]
Yeah, ] is just from the theme. It's a Quake style terminal and Quake uses ] as the prompt. Just a little bit of fun :)

**Comment by Massimo-B** [comment]
Is the :colours setting something that should be synced via Firefox sync? Apparently it does not sync.

**Comment by bovine3dom** [comment]
We removed automatic synchronisation of everything a couple of years ago. You can manually sync with `:firefoxsync{push,push}` but they don't merge settings so watch out - you might wipe your settings if you're not careful.

There's a caveat that custom themes aren't synchronised because usually they're too large for the 100kB quota Mozilla gives us.

**Comment by Massimo-B** [comment]
Nice hint. Where do I see which settings exactly are pushed via :firefoxsyncpush? Is that the same set of configuration that I would get from :mktridactylrc?

**Comment by Massimo-B** [comment]
I'm running fine now with `:colours quakelight` and it fixes the initial issue with the input line hidden by the statur bar.
However I don't feel comfortable with the Quake theme, because the input line in the top area doesn't feel vim-like and I never played Quake. More important: The input line is jumping when typing and the completion results above the input line are switching.

I prefer having the input line being below the completion results just like most vim forks do.
It seems to be possible to place the whole input line on a different position. Could you just change the current default theme and actually all themes that are placing the input line on the bottom to be just a few pixel above the Firefox status bar? Because having the Tridactyl input line hidden by the status bar actually is a bug and could be solved by placing it above, isn't it?

**Comment by bovine3dom** [comment]
I think that alternative would look very strange - did you try e.g. `:guiset hoverlink top-right` to move the pop-up instead?

**Comment by Massimo-B** [comment]
I tried all settings fro `hoverlink` and none of them made any difference when hovering a link. I also tried changing `statuspanel` and this doesn't change anything either when reloading a page.
My current Firefox theme is the integrated `Light` theme.
This is Firefox-91.9.1esr.

**Comment by bovine3dom** [comment]
Is `:native` installed and working?

**Comment by Massimo-B** [comment]
Yes, :native is installed, version 0.1.8
Today with a fresh restarted Firefox, the hoverlink is on top-right as expected. Thanks, everything is working fine now.

I updated :native to 0.3.6 now. Is :nativeupdate in the RC file for updating on every startup a good idea?

**Comment by finitemonkey** [comment]
I found this issue (and very little else) when searching for a solution to this problem and so here are some notes for future visitors to this page who find themselves in the same position.

I didn't want to install the native messenger so I searched for some other way to fix this, and found the following suggestion for `userChrome.css`:

```
#statuspanel[type="overLink"] {
      display:none !important;
}
```

However, after doing this, I discovered, to my surprise, that I actually eyeball that popup all the time and I was missing it!  (But this may work for you if you just don't want to see it at all, of course).

**For those who would like to keep the status bar and link destination popup, but move it out of the way of Tridactyl's UI**, I devised the following `userChrome.css`:

First, you may need to enable userChrome by setting the following in `about:config`:

`toolkit.legacyUserProfileCustomizations.stylesheets => true`

Then `<Mozilla profile dir>/chrome/userChrome.css` should look something like this:

```
@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");

/* Move Link Destination Overlay */
vbox.browserContainer {
        position: relative !important;
}

#statuspanel {
        position: absolute !important;
        top: 0px !important;
        margin-top: 0px !important;
}

#statuspanel-inner {
        height: auto !important;
}

#statuspanel-label {
        border-top-style: none !important;
        border-bottom-style: solid !important;
}

#statuspanel:not([mirror]) > #statuspanel-inner > #statuspanel-label:-moz-locale-dir(ltr),
#statuspanel[mirror] > #statuspanel-inner > #statuspanel-label:-moz-locale-dir(rtl) {
        border-top-right-radius: 0 !important;
        border-bottom-right-radius: .3em !important;
}
```

This seems to be working for me, I hope it helps somebody else too!

**Comment by Massimo-B** [comment]
This userChrome.css was working for me as a nice solution until I updated to 115.0.2. Now I see the status bar starting on top-left as expected, but covering all the space down to the border, actually half of my screen. Anybody noticed this as well?

**Comment by Massimo-B** [comment]
For completeness here is my complete userChrome.css:
[userChrome.css.txt](https://github.com/tridactyl/tridactyl/files/12216314/userChrome.css.txt)

**Comment by finitemonkey** [comment]
Here is an update for version 115.

@Massimo-B, would you give it a go and let me know if it works for you?

Tridactyl devs: Is there any chance that this could be subsumed into the Tridactyl css?  It seems absurd that users need to find an obscure post on an obscure Github issue and then enable userChrome just to have Tridactyl's UI not clash with Firefox's UI!

```diff
  @namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");

  /* Position status panel at top */
- vbox.browserContainer {
-     position: relative !important;
- }
- 
  #statuspanel {
      position: absolute !important;
      top: 0px !important;
-     margin-top: 0px !important;
+     padding: 0px !important;
+     height: fit-content !important;
  }
  
- #statuspanel-inner {
-         height: auto !important;
- }
- 
  #statuspanel-label {
-         border-top-style: none !important;
+         border-top-style: unset !important;
          border-bottom-style: solid !important;
  }
  
- #statuspanel:not([mirror]) > #statuspanel-inner > #statuspanel-label:-moz-locale-dir(ltr),
- #statuspanel[mirror] > #statuspanel-inner > #statuspanel-label:-moz-locale-dir(rtl) {
-         border-top-right-radius: 0em !important;
-         border-bottom-right-radius: .3em !important;
- }
+ #statuspanel:not([mirror]) > #statuspanel-label:-moz-locale-dir(ltr),
+ #statuspanel[mirror] > #statuspanel-label:-moz-locale-dir(rtl) {
+     border-top-right-radius: unset !important;
+     border-bottom-right-radius: 0.3em;
+ }
+ 
+ #statuspanel:not([mirror]) > #statuspanel-label:-moz-locale-dir(rtl),
+ #statuspanel[mirror] > #statuspanel-label:-moz-locale-dir(ltr) {
+     border-top-left-radius: unset !important;
+     border-bottom-left-radius: 0.3em;
+ }
```

**Comment by finitemonkey** [comment]
And the whole thing without the diff against the original:

```css
@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");

/* Position status panel at top */
#statuspanel {
    position: absolute !important;
    top: 0px !important;
    padding: 0px !important;
    height: fit-content !important;
}

#statuspanel-label {
    border-top-style: unset !important;
    border-bottom-style: solid !important;
}
  
#statuspanel:not([mirror]) > #statuspanel-label:-moz-locale-dir(ltr),
#statuspanel[mirror] > #statuspanel-label:-moz-locale-dir(rtl) {
    border-top-right-radius: unset !important;
    border-bottom-right-radius: 0.3em;
}

#statuspanel:not([mirror]) > #statuspanel-label:-moz-locale-dir(rtl),
#statuspanel[mirror] > #statuspanel-label:-moz-locale-dir(ltr) {
    border-top-left-radius: unset !important;
    border-bottom-left-radius: 0.3em;
}
```

**Comment by bovine3dom** [comment]
@finitemonkey 

> Is there any chance that this could be subsumed into the Tridactyl css?

We can't change Firefox CSS without the `:native` messenger installed.

If it is just an update to the `:guiset` code then yeah, we can and should include it.
